From b07d2f55fa39aa11c34eebe4e3340ad4844b61f5 Mon Sep 17 00:00:00 2001
From: Christian Poessinger <christian@poessinger.com>
Date: Tue, 4 May 2021 21:20:54 +0200
Subject: [PATCH] bgpd: when changing graceful-restart parameters, always
 return ok

vtysh will return an informational message to the user that changing any
graceful-shutdown related parameter either globally or per-neighbor/peer-group
will emit a message. This message is only informational for the user and should
not create a return code of 1 which signals "error"!

This fixes GitHub issue https://github.com/FRRouting/frr/issues/8403

Signed-off-by: Christian Poessinger <christian@poessinger.com>
---
 bgpd/bgp_vty.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/bgpd/bgp_vty.c b/bgpd/bgp_vty.c
index 40dd36812..a3b24c69e 100644
--- a/bgpd/bgp_vty.c
+++ b/bgpd/bgp_vty.c
@@ -2312,7 +2312,7 @@ DEFUN (bgp_graceful_restart,
 		zlog_debug("[BGP_GR] bgp_graceful_restart_cmd : END ");
 	vty_out(vty,
 		"Graceful restart configuration changed, reset all peers to take effect\n");
-	return bgp_vty_return(vty, ret);
+	return bgp_vty_return(vty, BGP_GR_SUCCESS);
 }
 
 DEFUN (no_bgp_graceful_restart,
@@ -2340,7 +2340,7 @@ DEFUN (no_bgp_graceful_restart,
 	vty_out(vty,
 		"Graceful restart configuration changed, reset all peers to take effect\n");
 
-	return bgp_vty_return(vty, ret);
+	return bgp_vty_return(vty, BGP_GR_SUCCESS);
 }
 
 DEFUN (bgp_graceful_restart_stalepath_time,
@@ -2496,7 +2496,7 @@ DEFUN (bgp_graceful_restart_disable,
 	vty_out(vty,
 		"Graceful restart configuration changed, reset all peers to take effect\n");
 
-	return bgp_vty_return(vty, ret);
+	return bgp_vty_return(vty, BGP_GR_SUCCESS);
 }
 
 DEFUN (no_bgp_graceful_restart_disable,
@@ -2526,7 +2526,7 @@ DEFUN (no_bgp_graceful_restart_disable,
 	vty_out(vty,
 		"Graceful restart configuration changed, reset all peers to take effect\n");
 
-	return bgp_vty_return(vty, ret);
+	return bgp_vty_return(vty, BGP_GR_SUCCESS);
 }
 
 DEFUN (bgp_neighbor_graceful_restart_set,
-- 
2.20.1

